### PR TITLE
CB-7477: Define API's for datalake and CB services to fasten the integration.

### PR DESCRIPTION
**Pull request by kkalvagadda1**
There is no functional change in this patch. It just defines the interface API's to request database backup/restore in datalake and CB services.

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java [hunk 1]
```diff
@@ -269,4 +273,17 @@ FlowIdentifier setClusterMaintenanceMode(@PathParam("workspaceId") Long workspac
...
+    @ApiOperation(value = DATABASE_BACKUP, nickname = "databaseBackup")
+    BackupV4Response backupDatabaseByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
+            @QueryParam("backupLocation") String backupLocation, @QueryParam("backupId") String backupId);
+
+    @POST
+    @Path("{name}/database_restore")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DATABASE_RESTORE, nickname = "databaseRestore")
+    RestoreV4Response restoreDatabaseByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
+            @QueryParam("backupLocation") String backupLocation, @QueryParam("backupId") String backupId);
```
Comment (pdarvasi) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java:288]: the same here.. is it uuid or name?

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java [hunk 2]
```diff
@@ -165,4 +169,27 @@ SdxClusterResponse create(@ValidStackNameFormat @ValidStackNameLength @PathParam
     @ApiOperation(value = "list advertised datalake versions", produces = MediaType.APPLICATION_JSON, nickname = "advertisedruntimes")
     List<AdvertisedRuntime> advertisedRuntimes();
 
+    @POST
+    @Path("{name}/backupDatabase")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "backup the database backing datalake ", produces = MediaType.APPLICATION_JSON, nickname = "backupDatabase")
+    SdxDatabaseBackupResponse backupDatabaseByName(@PathParam("name") String name, @QueryParam("backupId") String backupId, @QueryParam("backupLocation") String backupLocation);
```
Comment (pdarvasi) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java:176]: the same here.. is it uuid or name?

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseBackupStatusResponse.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.sequenceiq.sdx.api.model;
+
+public class SdxDatabaseBackupStatusResponse {
+    private String status;
```
Comment (pdarvasi) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseBackupStatusResponse.java:4]: what is the value set? If it is fixed then I would prefer using an enum, if not then it is fine.

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseRestoreStatusResponse.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package com.sequenceiq.sdx.api.model;
+
+public class SdxDatabaseRestoreStatusResponse {
+    private String status;
```
Comment (pdarvasi) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseRestoreStatusResponse.java:4]: what is the value set? If it is fixed then I would prefer using an enum, if not then it is fine.

File: datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java [hunk 5]
```diff
@@ -239,4 +243,27 @@ public FlowIdentifier stopByCrn(@ResourceCrn String crn) {
         return cdpConfigService.getAdvertisedRuntimes();
     }
 
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.DATALAKE_READ)
```
Comment (pdarvasi) [inline datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java:247]: Pls. introduce new AuthorizationResourceAction regarding backup/restore

**Comment by pdarvasi** [comment]
retest this please

**Comment by pdarvasi** [comment]
@kkalvagadda Pls. fix the failing tests to be able to merge

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java [hunk 6]
```diff
@@ -269,4 +273,17 @@ FlowIdentifier setClusterMaintenanceMode(@PathParam("workspaceId") Long workspac
     @ApiOperation(value = UPDATE_SALT, nickname = "updateSaltByName")
     FlowIdentifier updateSaltByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
 
+    @POST
+    @Path("{name}/database_backup")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DATABASE_BACKUP, nickname = "databaseBackup")
+    BackupV4Response backupDatabaseByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
+            @QueryParam("backupLocation") String backupLocation, @QueryParam("backupId") String backupId);
```
[the remark this answers is not in the document]
Comment (kkalvagadda1) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java:281]: backupid is the UUID for the backup. It is the Id generated by datalake-dr service.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java (hunk 1 above)
Comment (kkalvagadda1) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java:288]: backupId is the UUID for the backup. It is the Id generated by datalake-dr service. Yes, the name is the stack name. There is no use for backup name here.

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseBackupStatusResponse.java (hunk 3 above)
Comment (kkalvagadda1) [reply datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseBackupStatusResponse.java]: backupId is the UUID for the backup. It is the Id generated by datalake-dr service. Yes, the name is the stack name. There is no use for backup name here.

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java (hunk 2 above)
Comment (kkalvagadda1) [reply datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java:176]: backupId is the UUID for the backup. It is the Id generated by datalake-dr service. Yes, the name is the stack name. There is no use for backup name here.

File: datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java (hunk 5 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java]: done

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseRestoreStatusResponse.java (hunk 4 above)
Comment (kkalvagadda1) [reply datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxDatabaseRestoreStatusResponse.java]: done

**Pull request merged**